### PR TITLE
Avalonia UI 11 Preview 5

### PR DESCRIPTION
**Pull request by josephmoresena**
* Menu fixes for macOS.
* Nullable enabled.
* Fixes in ViewModel class names.
* ViewModel for MainControl as MainWindow state.
* Preserve window state when swapping Fluent and Simple themes.
* DataGrid fixes.

### CODE REVIEW

File: src/AvaloniaCoreRTDemo.csproj [hunk 1]
```diff
@@ -2,25 +2,31 @@
 
   <PropertyGroup>
     <!-- if you want to check exceptions, change this to "Exe" and start your app from the console -->
-    <OutputType>WinExe</OutputType>
+    <OutputType Condition="'$(Configuration.toUpper())' != 'DEBUG'">WinExe</OutputType>
+    <OutputType Condition="'$(Configuration.toUpper())' == 'DEBUG'">Exe</OutputType>
```
Comment (teobugslayer) [inline src/AvaloniaCoreRTDemo.csproj:6]: Why is this change needed? I prefer debug to be as close to release as possible, but maybe I am missing something?

File: src/Windows/ViewModels/AboutViewModel.cs [hunk 2]
```diff
@@ -8,11 +8,11 @@
 
 namespace AvaloniaCoreRTDemo.Windows.ViewModels
 {
-    internal record SystemDetail(string Key, string Value);
-    
+    internal record SystemDetail(String Key, String Value);
```
Comment (teobugslayer) [inline src/Windows/ViewModels/AboutViewModel.cs]: What's wrong with using C# keywords? I prefer to not introduce needless changes.

**Review by teobugslayer — CHANGES_REQUESTED** [review]
In addition to other comments, I'd prefer to leave the nuget.config intact. This project is often on the leading edge of either Avalonia or .NET and we need to modify it often enough.

Let's leave it with nuget sources commented out.

File: src/AvaloniaCoreRTDemo.csproj (hunk 1 above)
Comment (josephmoresena) [reply src/AvaloniaCoreRTDemo.csproj:6]: The project references to Avalonia.Diagnostics for debugging purposes only, I tried to extend that concept to IlcGenerateStackTraceData, StripSymbols and OutputType. That's useful for getting errors (especially those related to missing metadata), for once you resolve them, you may not need the console. When you use WinExe, there is no console in Windows, so these errors messages are lost by default.

File: src/Windows/ViewModels/AboutViewModel.cs (hunk 2 above)
Comment (josephmoresena) [reply src/Windows/ViewModels/AboutViewModel.cs]: I reverted this, as you point this change is needless. I made it just cause the whole class use System types instead C# alias. We never define nothing about that.

**Comment by josephmoresena** [comment]
> Let's leave it with nuget sources commented out.

Done!

File: src/AvaloniaCoreRTDemo.csproj (hunk 1 above)
Comment (ALIENQuake) [reply src/AvaloniaCoreRTDemo.csproj:6]: I agree that having a console window open when debugging is very useful. I don't need to separately launch the console and then launch an app in order to see what types are missing etc.

**Comment by teobugslayer** [comment]
Thank you @josephmoresena , much appreciated!
